### PR TITLE
Refine Add Commands

### LINKED ISSUE #85: Add and Delete any field with just the 'add' and 'delete' command

### PR DESCRIPTION
**Pull request by IsaacTin**

### CODE REVIEW

File: src/main/java/com/eva/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -149,9 +147,11 @@ public static Comment parseComment(String comment) throws ParseException {
...
             throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
         }
-        ArgumentMultimap argMultiMap = ArgumentTokenizer.tokenize(" " + comment, PREFIX_ADD,
-                PREFIX_DELETE, PREFIX_DATE, PREFIX_TITLE, PREFIX_DESC);
-        if (argMultiMap.getAllValues(PREFIX_DELETE).size() == 0) {
+        ArgumentMultimap argMultiMap = ArgumentTokenizer.tokenize(" " + comment,
+                PREFIX_DATE, PREFIX_TITLE, PREFIX_DESC);
+        if (argMultiMap.getAllValues(PREFIX_DATE).size() != 0
+                && argMultiMap.getAllValues(PREFIX_TITLE).size() != 0
+                && argMultiMap.getAllValues(PREFIX_DESC).size() != 0) {
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/logic/parser/ParserUtil.java:154]: maybe you can make use of the method arePrefixesPresent in parserutil

File: src/main/java/com/eva/logic/parser/DeleteCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        } else if (!deleteStaffCommand.isEmpty() && !deleteCommentCommand.isEmpty()) {
+            return new DeleteCommentCommandParser()
+                    .parse(" " + index.getOneBased() + " s- c- " + deleteCommentCommand.get());
+        } else if (!deleteStaffCommand.isEmpty()) {
+            return new DeleteStaffCommandParser()
+                    .parse(" " + index.getOneBased());
+        } else if (!deleteLeaveCommand.isEmpty()) {
+            return new DeleteLeaveCommandParser()
+                    .parse(" " + index.getOneBased()
+                            + " " + deleteLeaveCommand.get());
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/logic/parser/DeleteCommandParser.java:56]: i think we need to find a better way to handle this. I will come up with a app wide state to monitor what we are currently looking at so the commands dont mess up.

File: src/main/java/com/eva/logic/parser/AddCommandParser.java [hunk 3]
```diff
@@ -24,15 +34,73 @@
...
+        } else if (!addCommentCommand.isEmpty()) {
+            try {
+                index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            } catch (ParseException pe) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE_2), pe);
+            }
+            return new AddCommentCommandParser().parse(" " + index.getOneBased() + " c- " + addCommentCommand.get());
+        } else {
+            return this.addParse(args);
+        }
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/logic/parser/AddCommandParser.java:91]: same as DeleteCommandParser

File: src/main/java/com/eva/MainApp.java [hunk 4]
```diff
@@ -52,6 +52,7 @@
 
     @Override
     public void init() throws Exception {
+        assert false;
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/MainApp.java]: take this out if not the app wont start lol

**Review by Ben-Hanan — COMMENTED** [review]
Some changes to be made.

File: src/main/java/com/eva/MainApp.java (hunk 4 above)
Comment (TheSpaceCuber) [inline src/main/java/com/eva/MainApp.java]: Hi, should this be removed?

**Comment by TheSpaceCuber** [comment]
Okay I merge first! Thank Ben for comments.

**Pull request merged**